tests/open/24.t

```diff
@@ -2,17 +2,31 @@
 # vim: filetype=sh noexpandtab ts=8 sw=8
 # $FreeBSD: head/tools/regression/pjdfstest/tests/open/24.t 211352 2010-08-15 21:24:17Z pjd $
 
-desc="open returns EOPNOTSUPP when trying to open UNIX domain socket"
-
 dir=`dirname $0`
 . ${dir}/../misc.sh
 
+# POSIX doesn't explicitly state the errno for open(2)'ing sockets.
+case ${os} in
+Darwin|FreeBSD)
+	expected_error=EOPNOTSUPP
+	;;
+Linux)
+	expected_error=ENXIO
+	;;
+*)
+	echo "1..0 # SKIP: unsupported OS: ${os}"
```

Comment:
It sounds like you want to change the definition of `quick_exit` then?  Everywhere I've seen, it's used before any tests have been run, so it could use SKIP instead.